From 710790c7dc92941878fc89836119bf0d69beaecb Mon Sep 17 00:00:00 2001
From: Li Wang <li.wang@windriver.com>
Date: Tue, 11 Aug 2020 10:49:19 +0800
Subject: [PATCH] powerpc: Fix assembler warning about r0

The assembler says:
  arch/powerpc/platforms/85xx/sleep.S:1174: Warning: invalid register expression

It's objecting to the use of r0 as the RA argument. That's because
when RA = 0 the literal value 0 is used, rather than the content of
r0, making the use of r0 in the source potentially confusing.

Fix it to use a literal 0, the generated code is identical.

Signed-off-by: Li Wang <li.wang@windriver.com>
---
 arch/powerpc/platforms/85xx/sleep.S | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/powerpc/platforms/85xx/sleep.S b/arch/powerpc/platforms/85xx/sleep.S
index b7942edce2f0..b66e4382b3bf 100644
--- a/arch/powerpc/platforms/85xx/sleep.S
+++ b/arch/powerpc/platforms/85xx/sleep.S
@@ -1171,7 +1171,7 @@ _GLOBAL(fsl_booke_deep_sleep_resume)
 	li	r0, 0
 3:	mr	r3, r0
 	bl	loadcam_entry
-	addi	r0, r0, 1
+	addi	r0, 0, 1
 	bdnz	3b
 
 	/* restore return address */
-- 
2.17.1

